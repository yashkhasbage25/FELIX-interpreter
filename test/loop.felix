
var foo = fun (z) {
 	var y = z;
 	while( y > 0 )
 	{
 		println (" Square of num :", y," is ",y*y);
 		y = y-1;

 	}
};
foo(4);