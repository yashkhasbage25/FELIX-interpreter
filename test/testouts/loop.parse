[Var (GetID {tokenName = "foo"}) (Funct [GetID {tokenName = "z"}] [Var (GetID {tokenName = "y"}) (Id (GetID {tokenName = "z"})),While (Op > (Id (GetID {tokenName = "y"})) (Const (ConstDbl 0.0))) [Expr (Apply (Id (GetID {tokenName = "println"})) [Const (ConstStr " Square of num :"),Id (GetID {tokenName = "y"}),Const (ConstStr " is "),Op * (Id (GetID {tokenName = "y"})) (Id (GetID {tokenName = "y"}))]),Assign (GetID {tokenName = "y"}) (Op - (Id (GetID {tokenName = "y"})) (Const (ConstDbl 1.0)))]]),Expr (Apply (Id (GetID {tokenName = "foo"})) [Const (ConstDbl 4.0)])]
