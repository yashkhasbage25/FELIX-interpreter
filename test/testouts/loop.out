 Square of num :4.0 is 16.0
 Square of num :3.0 is 9.0
 Square of num :2.0 is 4.0
 Square of num :1.0 is 1.0
